### ISSUE TITLE
Optimization: Make tooltips cache

### ISSUE THREAD

**Issue by johnkslg** [StrangeLoopGames/EcoIssues#20280]
Tooltips are recalculated every time they're queried, and this can lead to a lot of expense on the server.
Make an option that can be set on the Tooltip attribute to 'cache'.  This will cache it on the first usage, and any subsequent calls to the tooltip will use that same value.  

This should be integrated into the tooltip system, so if a tooltip query is made for a given type it can cache the whole array of serialized bytes and just return that straight away for a given query.   The cache should accept the instance that the thing is being called on, if one exists.

For extra credit, can also make an 'update cache' ability, so some tooltips which are mostly-the-same can be 'dirtied' with calls from the code. 

Once this is done setup a few demos, and then make a separate task to go through all the tooltips to do it (can be given to someone else or even QA)

**Comment by johnkslg** [comment]
They should cache on the client actually to prevent any call to the server.
